### DISCUSSION TITLE
how to develop pantheon greeter

### DISCUSSION THREAD

**Discussion by jonbitzen** [elementary/greeter#522] [Q&A] [answered]
How do you guys do the test-debug iteration on the pantheon greeter?  The reason I ask is that I tried the bit at the bottom of the README, recommending lightdm --test-mode, and while it does start, I'm unable actually interact with it.  I gave that a go on 5.17/Hera, as well as Linux Mint 20.1 (which is my actual target).

I was going to fork the greeter to make some mods for a project I'm working on, and so I'm really looking for advice on best practices to iterate on changes so I don't have to re-invent that particular wheel.  I assume if lightdm test mode doesn't work "out of the box" either you have some way to make it work, or use some other method to test changes?

Thanks!

jonbitzen

**Comment by jonbitzen** [comment, marked as answer]
For posterity's sake, I've sorted a nice-enough system for myself.  I make a VM in VirtualBox with the target OS, and use shared folders to mount the host greeter source folder on the VM.  Then I compile it on the VM, but can edit on the host.  Additionally I ssh from the host into the guest with a terminal.  From there I can recompile on the command line, and let the guest OS sit at the login.  The upshot is that if I kill the greeter on the VM using the terminal login, it restarts the greeter on the guest almost-instantly, so you can test changes very quickly.

Hope this helps someone

**Reply by teamcons** [reply]
Thank you
